Remove when clock-gettime is fixed for clock_getcpuclockid.

See <http://lists.gnu.org/archive/html/bug-gnulib/2011-06/msg00227.html>
for details.

diff --git a/m4/clock_time.m4 b/m4/clock_time.m4
index 225355331..61373c4b7 100644
--- a/m4/clock_time.m4
+++ b/m4/clock_time.m4
@@ -25,8 +25,17 @@ AC_DEFUN([gl_CLOCK_TIME],
   AC_SUBST([LIB_CLOCK_GETTIME])
   gl_saved_libs=$LIBS
     AC_SEARCH_LIBS([clock_gettime], [rt posix4],
+                   [if test "$ac_cv_search_clock_gettime" = "none required"; then
+                      AC_SEARCH_LIBS([clock_getcpuclockid], [rt posix4],
+                                     [test "$ac_cv_search_clock_getcpuclockid" = "none required" \
+                                      || LIB_CLOCK_GETTIME=$ac_cv_search_clock_getcpuclockid],
+                                     [test "$ac_cv_search_clock_gettime" = "none required" \
+                                      || LIB_CLOCK_GETTIME=$ac_cv_search_clock_gettime])
+                    else
+                      LIB_CLOCK_GETTIME=$ac_cv_search_clock_gettime
+                    fi])
                    [test "$ac_cv_search_clock_gettime" = "none required" ||
                     LIB_CLOCK_GETTIME=$ac_cv_search_clock_gettime])
-    AC_CHECK_FUNCS([clock_getres clock_gettime clock_settime])
+    AC_CHECK_FUNCS([clock_getres clock_gettime clock_settime clock_getcpuclockid])
   LIBS=$gl_saved_libs
 ])
